### ISSUE TITLE
Site Editor: Browse styles expand indefinitely when a cover block is in template

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#57405]
### Description

If you add a cover block to a template in the Site Editor and open Styles > Browse Styles, the canvas will start to expand indefinitely.

### Step-by-step reproduction instructions

1. Open Appearance > Editor
2. Open any template and add a cover block.
3. Open Styles and select "browse styles".
4. Notice that the canvas starts to expand indefinitely.
5. Remove cover block.
6. Open Styles and select "browse styles". 
7. Notice problem goes away.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/26996883/d02e3e4d-bb64-4ae3-8e7c-f6906673cd96

### Environment info

- WP 6.4.2
- GB 17.2.3 & GB 17.3

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Similar issues have been reported several times in the past. We need to find a fundamental solution to this problem.

- #52297
- #49299
- #52092 
- #50449
- #49120

**Comment by annezazu** [comment]
Thanks for diving into the prior related issues :D Was going to do that today since I know I've opened a few already.

**Comment by ramonjd** [comment]
In WordPress 2023 theme it's noticeable on the home template with posts that have [featured images](https://github.com/WordPress/wordpress-develop/blob/8e88bc8efa31c194dd246084afb99d6a1929f245/src/wp-content/themes/twentytwentythree/templates/home.html#L12).

I hinted at it over in https://github.com/WordPress/gutenberg/issues/52297#issuecomment-1802929578 - if it's the CSS transition delay that's causing the race condition between height calculation and animation, I'd be open to ditching these transitions completely. E.g., 

```diff
diff --git a/packages/block-editor/src/components/iframe/index.js b/packages/block-editor/src/components/iframe/index.js
index 1939f75811..ddada79746 100644
--- a/packages/block-editor/src/components/iframe/index.js
+++ b/packages/block-editor/src/components/iframe/index.js
@@ -264,7 +264,8 @@ function Iframe( {
 						scale !== 1
 							? `scale( ${ scale } )`
 							: props.style?.transform,
-					transition: 'all .3s',
+					//transition: 'all .3s',
+					transition: 'none',
 				} }
 				ref={ useMergeRefs( [ ref, setRef ] ) }
 				tabIndex={ tabIndex }
```

**Comment by annezazu** [comment]
Ran into this again and 😭 -ed. This would be such a nice quality of life to have fixed.

**Comment by ramonjd** [comment]
I've tried to find a way to mitigate this again. Here's a brain dump of what I'm seeing.

The issue can be recreated by inserting some containing blocks with `vh` values set. I'm using the 2023 theme.

To test, I'm previewing the editor via "Browsing styles" in the site editor, which triggers the resizeObserver. Here's some test HTML:

```html
<!-- wp:group {"style":{"dimensions":{"minHeight":"100vh"}},"layout":{"type":"constrained"}} -->
<!-- wp:group {"style":{"dimensions":{"minHeight":"30.8vh"}},"layout":{"type":"constrained"}} -->
<div class="wp-block-group" style="min-height:30.8vh"><!-- wp:image {"id":37,"sizeSlug":"full","linkDestination":"none"} -->
<figure class="wp-block-image size-full"><img src="http://localhost:8888/wp-content/uploads/2024/01/white-bellied-swallow.jpg" alt="" class="wp-image-37"/></figure>
<!-- /wp:image --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->
```

From what I can see, the Iframe's [useResizeObserver](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/iframe/index.js#L126) is trying to calculate the height of the window as it's growing. 

Element "A",  with a `vh` minHeight value of `30vh` for example, will take up 30% of the viewport height. As the window height grows, Element "A" also grows because it wants to get to that 30%.

The [transition](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/iframe/index.js#L266) of `all .3s` makes it really obvious as it will stretch a little more slowly 😄 

The only way I could stop this was to restrict the zoomed out editor's height based on the iframe's surrounding container height.

- https://github.com/WordPress/gutenberg/pull/58484

I also turned off the transition because it's not helpful.

I'm not sure this is the right way though — it's a hack — just posting this in case other folks have ideas.
